presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -112,13 +112,21 @@ public CachingHiveMetastore(@ForCachingHiveMetastore HiveMetastore delegate, @Fo
                 config.getMetastoreCacheMaximumSize());
     }
 
-    public CachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Duration refreshInterval, long maximumSize)
+    public CachingHiveMetastore(HiveMetastore delegate, Executor executor, Duration cacheTtl, Optional<Duration> refreshInterval, long maximumSize)
     {
         this(
                 delegate,
                 executor,
                 OptionalLong.of(cacheTtl.toMillis()),
-                refreshInterval.toMillis() >= cacheTtl.toMillis() ? OptionalLong.empty() : OptionalLong.of(refreshInterval.toMillis()),
```

Comment:
I find it odd that `empty` means forever in `expiresAfterWriteMillis` and not "do not cache".

Shouldn't `empty` mean no caching and infinite be explicit (which we actually try to remove here)?

Reply:
> I find it odd that `empty` means forever in `expiresAfterWriteMillis` and not "do not cache".

And what should 0 mean in that case?

Reply:
Could you add a test case when `cacheTtl > refreshInterval` and the other way around? Do we need such test?

Reply:
I don't think we need it now.